src/rard/research/views/concordance.py

```diff
@@ -82,75 +212,144 @@ class ConcordanceCreateView(
     check_lock_object = "top_level_object"
 
     # create a concordance for an original text
-    model = Concordance
+    model = ConcordanceModel
     permission_required = ("research.add_concordance",)
-    fields = ("source", "identifier")
+    form_class = ConcordanceModelCreateForm
+    template_name = "research/concordancemodel_form.html"
 
     def dispatch(self, request, *args, **kwargs):
         # need to ensure we have the lock object view attribute
         # initialised in dispatch
         self.top_level_object = self.get_original_text().owner
-        if not isinstance(self.get_original_text().owner, Fragment) and not isinstance(
-            self.get_original_text().owner, AnonymousFragment
-        ):
-            raise Http404
 
         return super().dispatch(request, *args, **kwargs)
 
+    def get(self, request, *args, **kwargs):
+        edition_form = EditionForm()
+        concordance_form = None
+        return render(
+            request,
+            self.template_name,
+            context=self.get_context_data(edition_form, concordance_form),
+        )
+
+    def post(self, request, *args, **kwargs):
```

Comment:
I think you're trying to put too much functionality into one form here, which is why we end up with things like the error messages appearing when the second stage gets shown. Why not use separate edition create and concordance create forms with two separate views for each stage? For example:

* Edition view (stage 1): User selects an edition, or
* User selects create new edition and completes the relevant form
* User clicks next and either:
  * The concordance create view (stage 2) is called with a get request passing the selected edition as a query parameter, or
  * The edition create form is posted to the edition view, which creates the new edition and then the success url redirects to the concordance create view passing the newly created edition as a parameter
* The concordance create view returns a concordance form instance, passing the edition id in order to figure out the part identifier options
* On submit, the concordance view still handles both creating part_identifiers and concordances, but maybe split this functionality into separate functions to make the high level logic of the post function a bit clearer